addons/mail/models/mail_activity_template.py

```diff
@@ -0,0 +1,101 @@
+        "plan_id.res_model_ids". That's why this method is called by those models
+        when the mentioned fields are updated.
+        """
+        for record in self:
+            if record.plan_id.res_model_ids:
+                # Plan can only be applied to some model -> check that the activity types are compatible
+                if record.activity_type_id.res_model and (
+                        len(record.plan_id.res_model_ids) > 1 or
+                        record.plan_id.res_model_ids.model != record.activity_type_id.res_model):
+                    raise ValidationError(
+                        _('The activity type "%(activity_type_name)s" is not compatible with the plan "%(plan_name)s"'
+                          ' because the plan is limited to some model not compatible with "%(activity_type_name)s"'
+                          ' (only applicable to "%(activity_type_model)s").',
+                          activity_type_name=record.activity_type_id.name, plan_name=record.plan_id.name,
+                          activity_type_model=record.activity_type_id.res_model))
+            else:
+                # Plan can be applied to any model -> check that the activity types is not restrained to a model
+                if record.activity_type_id.res_model:
+                    raise ValidationError(
+                        _('The activity type "%(activity_type_name)s" is not compatible with the plan "%(plan_name)s"'
+                          ' because "%(activity_type_name)s" can only be applied to the model "%(activity_type_model)s"'
+                          ' while the plan can be applied to any model. Please limit the plan to the model '
+                          '"%(activity_type_model)s".',
+                          activity_type_name=record.activity_type_id.name, plan_name=record.plan_id.name,
+                          activity_type_model=record.activity_type_id.res_model))
+
+    @api.constrains('responsible_id', 'responsible')
+    def _check_responsible(self):
+        """ Ensure that responsible_id is set when responsible is set to "other". """
+        for record in self:
+            if record.responsible == 'other' and not record.responsible_id:
+                raise ValidationError(_('When selecting responsible "other", you must specify a responsible.'))
+
+    @api.depends('activity_type_id')
+    def _compute_default_summary(self):
+        for plan_type in self:
+            if plan_type.activity_type_id and plan_type.activity_type_id.summary:
+                plan_type.summary = plan_type.activity_type_id.summary
+            else:
+                plan_type.summary = False
+
+    @api.depends('responsible')
+    def _compute_responsible_id(self):
+        for record in self:
+            if record.responsible != 'other' and record.responsible_id:
+                record.responsible_id = False
+
+    def _get_responsible_id(self, on_demand_responsible, record):
```

Comment:
Mmmh ... naming could be improved to ease understand (it is a bit obscure :sweat_smile: ) maybe
```suggestion
    def _fetch_responsible_from_record(self, record, request_user_id):
```